internal/service/common/utils/persist.go

```diff
@@ -149,7 +150,44 @@ func PersistObjectWithChangeEvent[T db.Model](ctx context.Context, db *pgxpool.P
 		dataChangeEvent, err = PersistDataChangeEvent(
 			ctx, tx, record.TableName(), uuid, parentUUID, beforeModel, afterModel)
 		if err != nil {
-			return nil, fmt.Errorf("failed to persist resource type data change object: %w", err)
+			return nil, fmt.Errorf("failed to persist data change object: %w", err)
+		}
+	}
+
+	err = tx.Commit(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("failed to commit transaction: %w", err)
+	}
```

Comment:
There's also pgx.BeginFunc(...) helper func to basically the do the transaction. Might be a cleaner?

Reply:
fixed